src/main/java/com/openlattice/chronicle/services/message/MessageService.kt

```diff
@@ -0,0 +1,105 @@
+package com.openlattice.chronicle.services.message
+
+import com.google.common.collect.ArrayListMultimap
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.ListMultimap
+import com.openlattice.chronicle.constants.EdmConstants
+import com.openlattice.chronicle.constants.MessageDetails
+import com.openlattice.chronicle.constants.MessageOutcome
+import com.openlattice.chronicle.services.ApiCacheManager
+import com.openlattice.chronicle.services.edm.EdmCacheManager
+import com.openlattice.chronicle.services.enrollment.EnrollmentManager
+import com.openlattice.chronicle.services.entitysets.EntitySetIdsManager
+import com.openlattice.chronicle.services.twilio.TwilioManager
+import com.openlattice.chronicle.util.ChronicleServerUtil
+import com.openlattice.client.ApiClient
+import com.openlattice.data.DataAssociation
+import com.openlattice.data.DataGraph
+import org.slf4j.LoggerFactory
+import java.time.OffsetDateTime
+import java.util.*
+import java.util.concurrent.ExecutionException
+
+/**
+ * @author toddbergman &lt;todd@openlattice.com&gt;
+ */
+class MessageService(
+    private val apiCacheManager: ApiCacheManager,
+    private val edmCacheManager: EdmCacheManager,
+    private val enrollmentManager: EnrollmentManager,
+    private val entitySetIdsManager: EntitySetIdsManager,
+    private val twilioManager: TwilioManager
+) : MessageManager {
+    override fun sendMessage(
+        organizationId: UUID,
+        studyId: UUID,
+        participantId: String,
+        messageDetails: Map<String, String>
+    ) {
+        val message = MessageDetails(
+            messageDetails?.get("type") ?: "",
+            OffsetDateTime.now(),
+            participantId,
+            messageDetails?.get("phone") ?: "",
+            messageDetails?.get("url") ?: ""
+        )
+        val outcome = twilioManager.sendMessage(participantId, message)
+        try {
+            recordMessageSent(organizationId, studyId, outcome)
+            logger.info("Recorded notification {} sent", outcome.sid)
+        } catch (e: ExecutionException) {
+            logger.error("Unable to record notification sent for SID {}", outcome.sid, e)
+        }
+    }
+
+    @Throws(ExecutionException::class)
+    fun recordMessageSent(organizationId: UUID?, studyId: UUID?, messageOutcome: MessageOutcome) {
```

Comment:
studyId shouldn't be optional. organizationId can be optional if we are supporting legacy (I don't think we should though since are already set on doing away with legacy)